build/yarn_test.go

```diff
@@ -131,16 +131,24 @@ func TestCollectDepsForYarnProjectWithTraverse(t *testing.T) {
 	buildInfo, err := yarnBuild.ToBuildInfo()
 	assert.NoError(t, err)
 	assert.Len(t, buildInfo.Modules, 1)
-	validateModule(t, buildInfo.Modules[0], 1, 0, "build-info-go-tests:v1.0.0", entities.Npm, true)
-	assert.Equal(t, "json:9.0.6", buildInfo.Modules[0].Dependencies[0].Id)
-	assert.Equal(t, "test123", buildInfo.Modules[0].Dependencies[0].Sha1)
+	validateModule(t, buildInfo.Modules[0], 4, 0, "build-info-go-tests:v1.0.0", entities.Npm, true)
+	dependencies := []string{
```

Comment:
I switched the checking logic since we have 4 dependencies now instead of 1, and since they are kept in a map I cannot know the order in which they will be traversed. therefore I validated the correct amout of dependencies and checked if we have each of the dependencies I expect to have